cmake/Modules/G4BuildSettings.cmake

```diff
@@ -349,6 +349,15 @@ if(NOT BUILD_STATIC_LIBS AND NOT BUILD_SHARED_LIBS)
   message(FATAL_ERROR "Neither static nor shared libraries will be built")
 endif()
 
+if(MSVC)
+    option(GEANT4_MSVC_STATIC_RUNTIME "Use statically-linked multithreaded runtime library" OFF)
+    mark_as_advanced(GEANT4_MSVC_STATIC_RUNTIME)
+
+    if(GEANT4_MSVC_STATIC_RUNTIME)
+        set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+    endif()
+endif()
+
```

Comment:
Could you outline under what conditions the `GEANT4_MSVC_STATIC_RUNTIME` should be `OFF` and `ON` respectively please? Both for the cases of building Geant4, and for clients of the Geant4 libraries.

It would seem to have some relation to whether Geant4 static or DLLs are built, but I may be misunderstanding! I'm just wondering whether it should be automatically set, and/or only selected for one type of library.

Reply:
Our opinion is that it should be OFF by default. Mainly because this is Visual Studio's default.

At the moment, BUILD_STATIC_LIBS produces *almost* static libraries: they are still linked against the runtime libraries. Setting GEANT4_MSVC_STATIC_RUNTIME (maybe it should be renamed to GEANT4_MSVC_USE_STATIC_RUNTIME?) links against the static runtime library, and produces a fully static binary.

Visual Studio projects linked to Geant4 must use the same value for CMAKE_MSVC_RUNTIME_LIBRARY as was used to build Geant4.